### PR TITLE
Strip query strings and hashes when finding assets

### PR DESCRIPTION
**Pull request by lencioni**
Happo has some logic that looks for any local assets needed to render
the examples in a browser (e.g. images), and tries to bundle them up so
they can be available during screenshotting.

We've recently noticed a bug where some of our local images are failing
to load. When viewing source in the Happo UI, these images are 404ing,
and they do not exist in the assets zip file.

We noticed that these local images have some query strings added to
them. We believe that these query strings are causing these images to
not be bundled up with the other assets.

In this commit, I am hoping to make this better by stripping query
strings from the URLs. I started by trying to use node's URL built-in
parsing, but I found a couple of issues:

1. The `new URL()` API does not support relative URIs. There are some
   workarounds for this, like using a base of `relative:///`, but I was
   a bit worried about what unexpected consequences would arise from
   going this route. More information:
   https://github.com/nodejs/node/issues/12682
2. The `url.parse()` API is deprecated, and it gives you a URL object to
   work with. I wanted to essentially preserve the original format
   without adding any new information and I was a bit worried about
   getting that out of this object cleanly.

So in the end, I decided to write a simple regex for this.

My hunch is that this is an okay fix for now, but I imagine there might
be some situations where the query string is important for local images
and needs to be preserved in a better way so that all of the images are
correctly served. For example, imagine these two image URLs:

- ./foo.jpg?size=large
- ./foo.jpg?size=small

If these are used in an srcset, after this change, we will only bundle
up a single `./foo.jpg` image and that will be served for both.

I think the fix for this is a bit more involved (e.g. modifying the
image URLs so that the query strings are actually part of the filename,
or maybe storing them in a way that the server is able to pick the right
one when rendering examples for snapshotting), and it is unclear to me
if it actually matters at this time, so I'm calling YAGNI.

### CODE REVIEW

File: test/findCSSAssetPaths-test.js [hunk 1]
```diff
@@ -17,6 +17,11 @@ it('finds images', () => {
   expect(subject()).toEqual([{ assetPath: '/1x1.png', resolvePath: '/1x1.png' }]);
 });
 
+it('strips query strings', () => {
+  css = '.body { background-image: url("/1x1.png?pizza=yum");';
+  expect(subject()).toEqual([{ assetPath: '/1x1.png', resolvePath: '/1x1.png' }]);
```
Comment (lencioni) [inline test/findCSSAssetPaths-test.js]: @trotzig I wasn't entirely sure if the query string should be stripped for both assetPath and resolvePath

File: src/stripQueryString.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function stripQueryString(url) {
+  return url.replace(/\?.*$/, '');
```
Comment (ljharb) [inline src/stripQueryString.js]: you may also want to replace anchors, just in case:
```suggestion
  return url.replace(/[\?#].*$/, '');
```

Comment (lencioni) [reply src/stripQueryString.js]: Good idea! I'll add this

File: test/findCSSAssetPaths-test.js (hunk 1 above)
Comment (trotzig) [reply test/findCSSAssetPaths-test.js]: You did the right thing.

**Comment by trotzig** [comment]
Released in https://github.com/happo/happo.io/releases/tag/v3.25.2

**Pull request merged**